### DISCUSSION TITLE
Value 'sm_30' is not defined for option 'gpu-architecture'

### DISCUSSION THREAD

**Discussion by Qcellaris** [inducer/pycuda#337] [Troubleshooting] [answered]
Hi,

I installed the cuda toolkit and worked through the tutorial where the function doublify was created to double all entries in an array. Unfortunately, I can't get it working. Running this code results in the following error message:

nvcc fatal   : Value 'sm_30' is not defined for option 'gpu-architecture'

My setup:
Ubuntu 20.04
NVIDIA GeForce GTX 680
Driver Version: 470.82.01
CUDA Version: 11.4

Any clue if there is something wrong with my settings or is it something related to pycuda?

Kind regards,

**Comment by inducer** [comment, marked as answer]
I think the issue might be that Nvidia has removed support for the Kepler architecture from `nvcc` in the most recent versions. PyCUDA queries the GPU for its "compute capability", and it returned 3.0, which it then passes on to `nvcc`, which declared itself incapable of compiling for that compute capability. Downloading an older CUDA toolkit might help.

**Reply by Qcellaris** [reply]
Thank you for your support, I finally got it working. It took me several CUDA Toolkit installations before I found out that only CUDA Toolkits 9 and 10 support compute capability 3.0:

CUDA SDK 9.0 – 9.2 support compute capability 3.0 – 7.2
CUDA SDK 10.0 – 10.2 support compute capability 3.0 – 7.5

Hope this might help others that encounter this problem.
